yb-voyager/src/issue/issue.go

```diff
@@ -40,3 +43,42 @@ func (i Issue) IsFixedIn(v *ybversion.YBVersion) (bool, error) {
 	}
 	return v.GreaterThanOrEqual(minVersionFixedInSeries), nil
 }
+
+func (i Issue) GetImpact() string {
+	// Default value as IMPACT_LEVEL_1 if not defined in issue
+	return lo.Ternary(i.Impact != "", i.Impact, constants.IMPACT_LEVEL_1)
+}
+
+/*
+	Dynamic Impact Determination (TODO)
+	- We can define the impact calculator function based on issue type wherever/whenever needed
+	- Map will have functions only for issue type with dynamic impact determination
+
+	For example:
+
+	type ImpactCalcFunc func(issue QueryIssue, stats *PgStats) string
+
+	var impactCalculators = map[string]ImpactCalcFunc{
+		INHERITED_TABLE: inheritedTableImpactCalc,
+		// etc...
+	}
+
+	// Example dynamic function
+	func inheritedTableImpactCalc(i QueryIssue, stats *PgStats) string {
+		usage := stats.GetUsage(i.ObjectName) // e.g. how many reads/writes
+		if usage.WritesPerDay > 1000 {
+			return "LEVEL_2"
+		}
+		return "LEVEL_3"
+	}
+
+	// Update existing GetImpact() method
+	func (i Issue) GetImpact(stats *PgStats) string {
```

Comment:
I like the idea of the dynamic issue calculators!
It seems slightly off though that we will have to pass `pgStats` to `issue.GetImpact`.  Each issue type might need different inputs, so in the worst case, we will have to pass a lot of objects/args to `issue.GetImpact`; and we'll end up with a lot of nulls. 
THerefore, I think it might be better to have it as `getIssueImpact(Issue, pgStats, schemaInfo, ... )`. 

Basically, how about we look at it as each issue having a default impact level, and then based on additional info, we can modify it later?

Reply:
A follow up to this would be: do we then really need a getter func for GetImpact()? Can we just ensure we define it for every issue?

Reply:
> Can we just ensure we define it for every issue?

Yes we can, i added it to incorporate the possibility of dynamic impact determination, in future.

Reply:
> I don't have clarity on this yet. My assumption was that we would just need some stats to make that decision, which can be stored in a struct PgStats

It might well just be that, even I don't have clarity here. But I think even when we get to that, passing `pgStats` or any other input to `issue.getImpact` slightly seems weird. You'll have to write a big switch case on the issue Type inside `issue.getImpact()` and then calculate impact for each issue separately. 

Consider doing away with the getter for now since we don't have clarity on dynamic impact. When we actually get to dynamic impact, we can think of the design properly and then define it.

Reply:
we can keep the getter function for now. Its just one placeholder. We can change this later...

Reply:
If you're keeping the getter function, make the impact field unexported. It's conflicting/error-prone to have a getter as well as an exported field.

Reply:
removed the getter function.